cpp/src/Ice/SSL/SchannelTransceiverI.cpp

```diff
@@ -115,29 +131,37 @@ Schannel::TransceiverI::sslHandshake()
     SECURITY_STATUS err = SEC_E_OK;
     if (_state == StateHandshakeNotStarted)
     {
-        SCHANNEL_CRED cred = _engine->newCredentialsHandle(_incoming);
-        if (_localCertificateSelectionCallback)
+        _credentials = _localCredentialsSelectionCallback(_incoming ? _adapterName : _host);
+        if (_rootStore && !_credentials.hRootStore)
+        {
+            _credentials.hRootStore = CertDuplicateStore(_rootStore);
```

Comment:
The root store in the credentials is used to send the list of accepted CAs. _rootStore corresponds to the trusted CAs set with the authentication options or the IceSSL properties. Setting the credentials here offers a convenience, so the user doesn’t have to set it twice—once in the authentication options and again in the credentials.